test/functional/feature_config_args.py

```diff
@@ -332,5 +390,21 @@ def run_test(self):
         assert os.path.exists(os.path.join(new_data_dir_2, self.chain, 'blocks'))
 
 
+def get_temp_default_datadir(temp_dir: pathlib.Path) -> Tuple[dict, pathlib.Path]:
```

Comment:
Why do you need this os-specific path generator instead of just adding a single subdir to the direcotry where the test framework is already working? On my system at least, it doesn't actually return the "real" default path:

`/tmp/bitcoin_func_test_yg47qoyn/home/Library/Application Support/Bitcoin`

so why not just add `home/` on every platform and use that?

Reply:
Ok I understand now, I tried removing it but now I see the environment variables are important for setting the initial "where's the conf" directory without spoiling `-datadir`. I also notice `get_temp_default_datadir()` has a windows branch but windows will skip this test anyway! Not a bad idea to leave it in there anyway though I guess.

Reply:
Not a blocker for me, but curious if this kind of function should go in a broader utility package like `test_node.py` ?

Reply:
There is `get_datadir_path()` in _test_framework/util.py_:

https://github.com/bitcoin/bitcoin/blob/b759cefe936ed3991633acff215ea1dcec5ece28/test/functional/test_framework/util.py#L417-L418

So perhaps could go next to that in a future PR?